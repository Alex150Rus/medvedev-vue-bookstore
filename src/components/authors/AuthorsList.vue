<template>
  <v-expansion-panels>
    <v-expansion-panel
        v-for="(item, i) in authors"
        :key="i"
        class="js-listItem"
    >
      <v-expansion-panel-header>{{item.fullName}}</v-expansion-panel-header>
      <v-expansion-panel-content>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </v-expansion-panel-content>
    </v-expansion-panel>
  </v-expansion-panels>
</template>

<script>
import axios from 'axios';
export default {
<<<<<<< HEAD
name: "AuthorsList",
  data() {
    return {
      authors: []
    }
  },
  mounted() {
    axios.get('../json/authors.json')
        .then(response => this.authors = response.data)
  },
=======
name: "AuthorsList"
>>>>>>> parent of 63facc3... test work now))
}
</script>

<style scoped>

</style>