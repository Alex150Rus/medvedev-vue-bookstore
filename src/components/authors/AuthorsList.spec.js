import {shallowMount} from "@vue/test-utils";
import AuthorsList from "@/components/authors/AuthorsList";

import Vue from 'vue'
import Vuetify from 'vuetify'

Vue.use(Vuetify)

const wrapper = shallowMount(AuthorsList);
const vm = wrapper.vm;


describe('AuthorsList', () => {
    test('it mounts wo errors', () => {

        expect(wrapper.html()).toContain('v-expansion-panel-header')
    })
    test('quantity of list items to be equal to data items length', () => {
        const listItems = wrapper.findAll('.js-listItem');
        expect(listItems.length).toEqual(vm.data.items);
    })
})