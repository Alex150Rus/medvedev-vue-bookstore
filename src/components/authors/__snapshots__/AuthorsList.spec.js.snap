// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsList it mounts wo errors 1`] = `<div class="v-item-group theme--light v-expansion-panels"></div>`;
